tests/testthat/test-bind-rows.R

```diff
@@ -145,6 +145,17 @@ test_that("bind_rows() handles named list", {
   expect_equal(bind_rows(x), tibble(x = 1, y = 2, z = 3))
 })
 
+test_that("bind_rows() handles empty names in a list", {
+  x <- lapply(1:5, function(x) mtcars)
+  names(x) = paste0("id_", 1:5)
+  names(x)[c(3, 5)] = NA_character_
+  x <- bind_rows(x, .id = "id")
+  x_ids <- x$id |> unique()
+
+  # If names are missing, bind_rows will replace with index#
+  expect_equal(x_ids, c("id_1", "id_2", "3", "id_4", "5"))
```

Comment:
```suggestion
  x <- rep(list(data.frame(x = 1)), times = 5)
  names(x) <- paste0("id_", 1:5)
  names(x)[c(3, 5)] <- NA_character_
  x <- bind_rows(x, .id = "id")

  # If names are missing, bind_rows will replace with index
  expect_identical(x$id, c("id_1", "id_2", "3", "id_4", "5"))
```

A few things here (none of which we'd expect you to know ahead of time!):

- Using `rep(list(data.frame(x = 1)), times = 5)` because `rep()` is a little clearer and `data.frame(x = 1)` creates a 1 row data frame, so you don't need `x_ids <- x$id |> unique()` anymore
- Consistently using `<-` rather than `=`
- Fixed a typo, `index#` -> `index`
- `expect_identical()` over `expect_equal()`, it's a little stricter!

Reply:
Ahh! some copy and paste errors on my part.
But this is all really great info. Thank you so much for sharing. Definitely taking notes 😅